src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -1806,9 +1808,18 @@ export const handleUpdateLogic = [
           .items(
             Joi.object({
               field: Joi.string().required(),
-              state: Joi.string().required(),
-              value: Joi.string().required(),
-              ifValueType: Joi.string(),
+              state: Joi.string()
+                .valid(...Object.values(LogicConditionState))
+                .required(),
+              value: Joi.alternatives()
+                .try(
+                  Joi.number(),
+                  Joi.string(),
+                  Joi.array().items(Joi.string()),
+                  Joi.array().items(Joi.number()),
+                )
+                .required(),
+              ifValueType: Joi.valid(...Object.values(LogicIfValue)),
```

Comment:
can I check why this is not required, and should it be? also is the behaviour different if you exclude `Joi.string()`, as in the `state` key?

Reply:
thanks, made it required. Excluding `string()` shouldn't make a difference because all the values in LogicIfValue are strings, but I think adding `string()` is more [canonical](https://joi.dev/api/?v=17.4.0#anyvalidvalues---aliases-equal). Edited accordingly